ledger/src/blockstore_meta.rs

```diff
@@ -526,6 +526,19 @@ impl OptimisticSlotMetaVersioned {
         }
     }
 }
+
+#[derive(Clone, Copy, Debug, Eq, PartialEq, Serialize, Deserialize)]
+pub struct SizeDetails {
```

Comment:
How would you expect to populate these fields?
When you receive a (code or data) shred, you do not know if that belongs to the last FEC set unless it is the very last data shred.

Reply:
we wait until we receive a shred with `LAST_SHRED_IN_SLOT` flag. At that point we know the `fec_set_index` of the last fec set.
If we have already seen a coding shred at that point we can use the `ErasureMeta` to perform a one time scan of blockstore to see how many coding/data shreds we have received. 
For further shred insertions, we increment `num_data_received` or `num_coding_received`.

Reply:
>  If we have already seen a coding shred at that point we can use the ErasureMeta to perform a one time scan of blockstore to see how many coding/data shreds we have received.

and what if we have not received a coding shred yet?

Reply:
> if not we we set the num_data_inserted and num_coding_inserted to 0.
If we receive a coding shred while num_coding_inserted is 0, we perform the one time scan using the ErasureMeta and update